From 966926e3db5ea06b962fac386fa3c7223f9a14fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gonzalo=20Tornar=C3=ADa?= <tornaria@cmat.edu.uy>
Date: Tue, 21 Dec 2021 14:57:20 -0300
Subject: notebook: fix deprecation warning

While doctesting `src/sage/repl/ipython_kernel/install.py`:

> DeprecationWarning: invalid escape sequence '\s'

The solution is to make the regex a raw string.
---
 .../f99beefdf45dd2891bb7ebdd93f7a67e44623aee.patch | 27 ++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 build/pkgs/notebook/patches/f99beefdf45dd2891bb7ebdd93f7a67e44623aee.patch

diff --git a/build/pkgs/notebook/patches/f99beefdf45dd2891bb7ebdd93f7a67e44623aee.patch b/build/pkgs/notebook/patches/f99beefdf45dd2891bb7ebdd93f7a67e44623aee.patch
new file mode 100644
index 00000000..29875b0
--- /dev/null
+++ b/build/pkgs/notebook/patches/f99beefdf45dd2891bb7ebdd93f7a67e44623aee.patch
@@ -0,0 +1,27 @@
+From f99beefdf45dd2891bb7ebdd93f7a67e44623aee Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Gonzalo=20Tornar=C3=ADa?= <tornaria@cmat.edu.uy>
+Date: Tue, 21 Dec 2021 14:54:24 -0300
+Subject: [PATCH] Fix deprecation warning
+
+In sagemath, while doctesting `src/sage/repl/ipython_kernel/install.py`:
+
+> DeprecationWarning: invalid escape sequence '\s'
+
+The solution is to make the regex a raw string.
+---
+ notebook/auth/login.py | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/notebook/auth/login.py b/notebook/auth/login.py
+index 1ac434dc5e..16c4e7a356 100644
+--- a/notebook/auth/login.py
++++ b/notebook/auth/login.py
+@@ -112,7 +112,7 @@ def set_login_cookie(cls, handler, user_id=None):
+         handler.set_secure_cookie(handler.cookie_name, user_id, **cookie_options)
+         return user_id
+ 
+-    auth_header_pat = re.compile('token\s+(.+)', re.IGNORECASE)
++    auth_header_pat = re.compile(r'token\s+(.+)', re.IGNORECASE)
+ 
+     @classmethod
+     def get_token(cls, handler):
-- 
cgit v1.0-1-gd88e

